### ISSUE TITLE
Question: Should compilation with QPU=1 be blocked on non-RPi platforms?

### ISSUE THREAD

**Issue by wimrijnders** [mn416/QPULib#18]
Out of curiosity, I compiled with `make QPU=1` on my 64-bit Intel machine. I got type errors:

```
Lib/VideoCore/SharedArray.h:121:14: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     gpu_base = (void*) mem_lock(mb, handle);
```

This shouldn't be working, of course. The question is: how to deal with this?

My personal preference is to block this with an error, indicating that it can only work on RPi (which is 32-bits). So it actually shouldn't be compiling at all. Can you agree with this?

A lesser point is: irrespective of this error, should the compile be possible? This is sort of a moot thing, but for completeness sake it might be worth considering.

**Comment by mn416** [comment]
I don't think it's expected to compile, so am happy for an error message to be raised on non-Pi platforms, assuming you can reliably determine a non-Pi platform.

**Comment by wimrijnders** [comment]
Resolved by #38.
